js/calendar.js

```diff
@@ -1,43 +1,39 @@
-export function renderCalendarCells() {
-    document.addEventListener('DOMContentLoaded', () => {
-        const calendarCells = document.getElementById('calendarCells');
-        const cellsInOneColumn= 19;
-        const daysInWeek = 7;
-        for (let i = 0; i < cellsInOneColumn * daysInWeek; i++) {
-            const cell = document.createElement('div');
-            cell.className = 'cell';
-            calendarCells.appendChild(cell);
-        }
-    });
-}
-function getStartOfWeek(date) {
-    const day = date.getDay(); 
-    const diff = date.getDate() - day + (day === 0 ? -6 : 1); 
-    return new Date(date.setDate(diff));
-}
-function isToday(date) {
-    const today = new Date();
-    return date.getDate() === today.getDate() &&
-           date.getMonth() === today.getMonth() &&
-           date.getFullYear() === today.getFullYear();
+import { getStartOfWeek, isToday } from "./utils.js";
+
+export let startDayOfWeek = new Date();
+export function renderCalendarCells(loadEventsCallback) {
+  document.addEventListener("DOMContentLoaded", () => {
+    const calendarCells = document.getElementById("calendarCells");
+    const cellsInOneColumn = 19;
+    const daysInWeek = 7;
+    for (let i = 0; i < cellsInOneColumn * daysInWeek; i++) {
+      const cell = document.createElement("div");
+      cell.className = "cell";
+      calendarCells.appendChild(cell);
+    }
+    if (typeof loadEventsCallback === "function") {
+      loadEventsCallback();
+    }
+  });
```

Comment:
let's start splitting to smaller functions, so that one function does pretty much one thing

```suggestion
export function renderMainCalendar(date) {
  document.addEventListener("DOMContentLoaded", () => {
    // we could also move week header rendering here as well
    renderCalendarCells(date)
    loadWeekEvents(date)
  });
```

why did you need check typeof loadEventsCallback === "function" ?